### ISSUE TITLE
Crash: "Shutdown before Coda instance was created"

### ISSUE THREAD

**Issue by dguess** [MinaProtocol/mina#3520]
Trying to get Coda running for the first time on macOS Mojave (Version 10.14.6)and getting the following crash report: 

> **$ coda daemon -peer filet-mignon.o1test.net:8303**
> 2019-09-27 02:15:19 UTC [Info] Coda daemon is booting up; built with commit "c43d954f30a3b45c23ff31bba9436a2e14b56483" on branch "release/0.0.8-beta"
> 2019-09-27 02:15:19 UTC [Info] Daemon will expire at "2019-10-08 14:00:00-07:00"
> 2019-09-27 02:15:19 UTC [Info] Booting may take several seconds, please wait
> 2019-09-27 02:16:28 UTC [Info] Daemon child process 1340 terminated with nonzero exit code 1
> 2019-09-27 02:16:28 UTC [Fatal] Unhandled top-level exception: $exn
> Generating crash report
> 	exn: "(monitor.ml.Error\n  (Unix.Unix_error \"No child processes\" waitpid\n    \"((mode (WNOHANG)) (pid 1340))\")\n  (\"Raised at file \\\"src/core_unix.ml\\\", line 50, characters 4-43\"\n    \"Called from file \\\"src/core_unix.ml\\\", line 930, characters 4-246\"\n    \"Called from file \\\"src/unix_syscalls.ml\\\", line 596, characters 10-39\"\n    \"Called from file \\\"src/unix_syscalls.ml\\\" (inlined), line 602, characters 21-55\"\n    \"Called from file \\\"src/unix_syscalls.ml\\\", line 610, characters 33-48\"\n    \"Called from file \\\"camlinternalLazy.ml\\\", line 27, characters 17-27\"\n    \"Re-raised at file \\\"camlinternalLazy.ml\\\", line 34, characters 4-11\"\n    \"Called from file \\\"src/process.ml\\\" (inlined), line 142, characters 13-25\"\n    \"Called from file \\\"external/rpc_parallel/src/parallel.ml\\\", line 1206, characters 14-34\"\n    \"Called from file \\\"src/deferred0.ml\\\", line 56, characters 64-69\"\n    \"Called from file \\\"src/job_queue.ml\\\" (inlined), line 131, characters 2-5\"\n    \"Called from file \\\"src/job_queue.ml\\\", line 170, characters 6-47\"\n    \"Caught by monitor coda\"))"
>
>
>  ☠  Coda daemon crashed. The Coda Protocol developers would like to know why!

[coda_crash_report_2019-09-27_02-16-28.768216.tar.gz](https://github.com/CodaProtocol/coda/files/3660591/coda_crash_report_2019-09-27_02-16-28.768216.tar.gz)

**Comment by enolan** [comment]
Hmm. It failed to start one of the worker processes. How much RAM does your
machine have?

**Comment by dguess** [comment]
@enolan it has 16GB of RAM.

I've run it again on the same machine with a different internet connection and it works fine so it seems as even though `sudo upnpc -r 8302 TCP 8303 UDP` worked fine that port is still being blocked.

**Comment by dguess** [comment]
Reopening after discussion with @o1pranay. I am able to reproduce this issue on two different routers now.

**Comment by o1pranay** [comment]
cc @psteckler who I think fixed this one

**Comment by bkase** [comment]
Just got a WNOHANG error locally

**Comment by bkase** [comment]
[coda_crash_report_2019-10-10_00-17-24.949236.tar.gz](https://github.com/CodaProtocol/coda/files/3709920/coda_crash_report_2019-10-10_00-17-24.949236.tar.gz)

**Comment by bkase** [comment]
this was after running for 1h42min

**Comment by bkase** [comment]
It seemed to happen immediately after I sent a transaction

**Comment by psteckler** [comment]
It must be from a `wait` other than the one I wrapped in `try_with`. I was not certain that was the source of the error.

@o1pranay said he was able to reproduce this easily. If so, maybe we can track this down.

**Comment by dguess** [comment]
FYI I'm still getting this issue on my local macs when trying to run the new build:

```coda daemon \
    -discovery-port 8303 \
    -peer /ip4/52.39.56.50/tcp/8303/ipfs/12D3KooWHMmfuS9DmmK9eH4GC31arDhbtHEBQzX6PwPtQftxzwJs \
    -peer /ip4/18.212.230.102/tcp/8303/ipfs/12D3KooWAux9MAW1yAdD8gsDbYHmgVjRvdfYkpkfX7AnyGvQaRPF \
    -peer /ip4/52.13.17.206/tcp/8303/ipfs/12D3KooWCZA4pPWmDAkQf6riDQ3XMRN5k99tCsiRhBAPZCkA8re7 \
    -propose-key keys/my-wallet```

**Comment by aneesharaines** [comment]
related 3.2b crashes
#5050

**Comment by mrmr1993** [comment]
This is likely fixed by #7611. Closing.
